src/classes/RD_RecurringDonations.cls

```diff
@@ -795,77 +766,391 @@ public class RD_RecurringDonations {
-                updateExistingOpps(reevaluateopps, dmlWrapper);
-            else
-                updateExistingOppsFuture(reevaluateopps);
+
+        if (!openEndedRDsToRecalculate.isEmpty()) {
+            List<Opportunity> openOpps = [SELECT Id
+                    , Amount
+                    , Name
+                    , CampaignId
+                    , AccountID
+                    , CloseDate
+                    , StageName
+                    , Recurring_Donation_Installment_Number__c
+                    , npe03__Recurring_Donation__c
+            FROM Opportunity
+            WHERE npe03__Recurring_Donation__c IN :openEndedRDsToRecalculate
+            AND isClosed = false
+            ORDER BY CloseDate ASC];
+            handleOpenEndedScheduleRecalculation(openEndedRDsToRecalculate, openOpps, dmlWrapper);
         }
-        //needs a total reinsert, not an update
-        if (!reinsertopps.isEmpty()){
-            //clear existing open opps and recreate
-            delete [select id from Opportunity where npe03__Recurring_Donation__c IN :reinsertopps and isClosed = false];
-            if (reinsertopps.size() == 1 || system.isFuture() || system.isBatch()){
-                insertOppsOnRecurringDonationInsert(reinsertopps);                  
+
+        if (!fixedLengthRDsToRecalculate.isEmpty()) {
+            List<Opportunity> opps = [SELECT Id
+                    , Amount
+                    , Name
+                    , CampaignId
+                    , AccountID
+                    , CloseDate
+                    , StageName
+                    , isWon
+                    , Recurring_Donation_Installment_Number__c
+                    , npe03__Recurring_Donation__c
+            FROM Opportunity
+            WHERE npe03__Recurring_Donation__c IN :fixedLengthRDsToRecalculate
+            ORDER BY CloseDate ASC];
+            handleFixedLengthScheduleRecalculation(fixedLengthRDsToRecalculate, opps, dmlWrapper);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Re-evaluates all open Opportunities of an Open-Ended Recurring Donation when it is updated to require schedule recalculation.
+    * @param recurringDonations The list of RDs to process
+    * @param openOpportunities The list of open Opportunities to process
+    * @param dmlWrapper Object to hold the Opps that need deleting or updating
+    * @return void
+    */
+    private static void handleOpenEndedScheduleRecalculation(List<npe03__Recurring_Donation__c> recurringDonations
+            , List<Opportunity> openOpportunities
+            , TDTM_Runnable.DmlWrapper dmlWrapper) {
+
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
+
+        //build a map of recurring donation ID and opps
+        Map<Id, List<Opportunity>> RDOppMap = new Map<Id, List<Opportunity>>();
```

Comment:
For new code, all vars should start with a lowercase. I know this class is pretty inconsistent with that, but anything new we write or fix should use camelcase vars.

Reply:
Since this is new code, can you update the var/type first letter case on these?